Session-4/CurrencyExchangeApp/CurrencyConverter/CurrencyConverter.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Configuration;
+using RestSharp;
+using System.Collections.Generic;
+using System.IO;
+
+
+namespace CurrencyConverter
+{
+    public class CurrencyExchange
+    {
+        public void GetApidata()
+        {
+            try {
+                RestClient client = new RestClient("https://api.apilayer.com/currency_data/live?source=INR&currencies=");
+                RestRequest request = new RestRequest();
+                request.AddHeader("apikey", "NKJlyNE9umLraoNvP87N7W9v6NNePbI8");
+                ApiResponse apiResponse = JsonConvert.DeserializeObject<ApiResponse>(client.Get(request).Content);
+                apiResponse.quotes.Add("timestamp", apiResponse.timestamp);
+                string path = System.Configuration.ConfigurationManager.AppSettings["path"];
+                File.WriteAllText(path, JsonConvert.SerializeObject((object)apiResponse.quotes));
+            }
+            catch(Exception e) {
+                throw e.InnerException;
+            }
+                
+        }
+
+        public double Convert(string fromCurrency, string toCurrency, double Currencyvalue)
+        {
+            CurrrencyNotFoundException notFoundException = new CurrrencyNotFoundException("Enter a valid Currency[Ex : USD, INR]");
+            if (toCurrency == fromCurrency)
+                return 1.0;
+            string key = fromCurrency + toCurrency;
+            if (File.Exists("C:\\Users\\prashanth yarram\\OneDrive\\Desktop\\sample.txt"))
+            {
+                DateTime dateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0).AddSeconds((double)(int)JsonConvert.DeserializeObject<Dictionary<string, double>>(File.ReadAllText("C:\\Users\\prashanth yarram\\OneDrive\\Desktop\\sample.txt"))["timestamp"]);
+                dateTime = dateTime.ToLocalTime().Date;
+                string str1 = dateTime.ToString("M/d/yyyy");
+                dateTime = DateTime.Now;
+                string str2 = dateTime.ToString("M/d/yyyy");
+                if (str1 != str2)
+                    this.GetApidata();
+            }
+            else
+                this.GetApidata();
+            Dictionary<string, double> dictionary = JsonConvert.DeserializeObject<Dictionary<string, double>>(File.ReadAllText("C:\\Users\\prashanth yarram\\OneDrive\\Desktop\\sample.txt"));
+            if (toCurrency == "INR")
```

Comment:
what is the purpose of this condition? your code should work for any type of currencies. (Eg: USD to GBP)

Reply:
The purpose of this line is using that the API that I am using gets currency exchange rates based on only one currency. So here I am using INR as my base currency and I get exchange rates in Currency to INR format so to convert a non INR  value Ex: USD to GBP I have to convert the USD to INR and then from INR to GBP.